pkg/codegen/schema/schema.go

```diff
@@ -1417,6 +1422,8 @@ func bindSpec(spec PackageSpec, languages map[string]Language, loader Loader) (*
 		} else {
 			version = &v
 		}
+	} else {
+		diags = diags.Append(errorf("#/version", "No version provided"))
```

Comment:
This will not work without downstream code changes. We allow schemas to elide the version argument to handle situations where the version is not known at the time the schema is generated, which is the case in ~all of our providers. All of our providers pull their version from a git tag, which is not available at the time the schema is generated; it's complicated.

Reply:
This flatly contradicts what the [docs](https://www.pulumi.com/docs/guides/pulumi-packages/schema/) say about the `package.version` property. Is this a distinction we make for our own packages, or a general rule (in which case we should update the docs)?

Reply:
Issue in Hugo here: https://github.com/pulumi/pulumi-hugo/issues/570

Reply:
> Is this a distinction we make for our own packages

This, basically. I think it's okay to leave the documentation as-is.